### PR TITLE
Update gradle version for Android Studio 2.3

### PR DESCRIPTION
**Pull request by jonathan-caryl**
[Android Studio 2.3 has been released](https://android-developers.googleblog.com/2017/03/android-studio-2-3.html) and it wants you to upgrade the gradle version used in the project.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Wed Dec 14 12:53:56 CET 2016
+#Fri Mar 03 09:38:24 GMT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.2.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (vRallev) [inline gradle/wrapper/gradle-wrapper.properties]: Use 3.4

Comment (jonathan-caryl) [inline gradle/wrapper/gradle-wrapper.properties]: If I change it to 

    distributionUrl=https\://services.gradle.org/distributions/gradle-3.4-all.zip

then the build fails like so:

```* What went wrong:
A problem occurred configuring project ':plugin'.
> Could not resolve all dependencies for configuration ':plugin:_debugPublish'.
   > Exception thrown while executing model rule: PublishingPlugin.Rules#publishing(ExtensionContainer)
      > org/gradle/api/internal/component/Usage
```

Is there much advantage to upgrading beyond what Android Studio prompts you to do automatically?

Comment (vRallev) [inline gradle/wrapper/gradle-wrapper.properties]: Yes, 3.4 provides significant improvements: https://docs.gradle.org/current/release-notes.html But there seems to be an issue with the publishing plugin. Not sure about that though.

Comment (passsy) [inline gradle/wrapper/gradle-wrapper.properties]: Please use `./gradlew wrapper --gradle-version 3.3 --distribution-type all` to upgrade the wrapper. This also upgrades the `gradle-wrapper.jar`

File: build.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
```
Comment (passsy) [inline build.gradle]: I'm not a fan up upgrading dependencies on day 1. Being always on the latest version is also not required for this project.

**Comment by StefMa** [comment]
I totally agree with the comments from @passsy .

@jonathan-caryl are you still working on it?

**Comment by jonathan-caryl** [comment]
@StefMa I've updated the `gradle-wrapper.jar` as requested by @passsy and rebased.

File: build.gradle (hunk 2 above)
Comment (StefMa) [inline build.gradle]: It seems that 2.3.0 don't break something.
I've also nothing read about "2.3.0 the horror release" (or similar 😺)
So I approve it and merge it 🎉

**Comment by StefMa** [comment]
@jonathan-caryl thank you for the contribution 👍

**Pull request merged**